tests/unit/preprocessor/_volume/test_volume.py

```diff
@@ -78,6 +79,70 @@ def setUp(self):
         iris.util.guess_coord_axis(self.grid_4d.coord('zcoord'))
         iris.util.guess_coord_axis(self.grid_4d_2.coord('zcoord'))
 
+    def test_axis_statistics(self):
+        """Test axis statistics in multiple operators. """
+        for operator in ['mean', 'median', 'min', 'max', 'rms']:
+            result = axis_statistics(self.grid_4d, 'z', operator)
+            expected = np.ma.ones((2, 2, 2))
+            self.assert_array_equal(result.data, expected)
+
+        for operator in ['std_dev', 'variance']:
+            result = axis_statistics(self.grid_4d, 'z', operator)
+            expected = np.ma.zeros((2, 2, 2))
+            self.assert_array_equal(result.data, expected)
+
+        result = axis_statistics(self.grid_4d, 'z', 'sum')
+        expected = np.ma.ones((2, 2, 2)) * 250
+        self.assert_array_equal(result.data, expected)
```

Comment:
This test is carefully set up such that the result is the same for many different operations. This is a bit dangerous, imho, since it will miss many types of errors, such as wrong selection of operator, mixing of wrong axis, etc.
It would be better to have slightly different input (e.g. from linrange) and unique outputs. The programming effort can be controlled using paramtrized tests, see https://docs.pytest.org/en/6.2.x/parametrize.html. This way, even the two loops can be removed in favor of a single one of their bodies and a list of `(operator, expected) pairs for the parametrization.

Reply:
I had some issues when trying to parametrize a test inside a class, so I wrote it in a loop

Reply:
Could you try it one more time? If that doesn't work, we'll call the resulting method `_inner_test_something` and implement the loop explicitly in a proper test method of the class.

Regardless of the approach, the point about the dangerous tests stands.

Reply:
Yes, at the end I went for writing each test case separately. Hope that works for the time being.

Reply:
Looks nice and matches what is done in https://github.com/ESMValGroup/ESMValCore/blob/4a840fb76d89453643758210f3aab9ad5caa4632/tests/unit/preprocessor/_time/test_time.py#L526 (can be changed another time)